pkgs/development/tools/build-managers/apache-maven/default.nix

```diff
@@ -1,21 +1,43 @@
-{ lib, stdenv, fetchurl, jdk, makeWrapper }:
+{ lib
+, stdenvNoCC
+, fetchurl
+, jdk
+, makeWrapper
+, callPackage
+}:
 
 assert jdk != null;
 
-stdenv.mkDerivation rec {
+stdenvNoCC.mkDerivation (finalAttrs: {
   pname = "apache-maven";
   version = "3.9.2";
 
-  builder = ./builder.sh;
-
   src = fetchurl {
-    url = "mirror://apache/maven/maven-3/${version}/binaries/${pname}-${version}-bin.tar.gz";
-    sha256 = "sha256-gJ7zIgxtF5GVwGwyTLmm002Oy6Vmxc/Y64MWe8A0EX0=";
+    url = "mirror://apache/maven/maven-3/${finalAttrs.version}/binaries/${finalAttrs.pname}-${finalAttrs.version}-bin.tar.gz";
+    hash = "sha256-gJ7zIgxtF5GVwGwyTLmm002Oy6Vmxc/Y64MWe8A0EX0=";
   };
 
+  sourceRoot = ".";
+
   nativeBuildInputs = [ makeWrapper ];
 
-  inherit jdk;
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/maven
+    cp -r ${finalAttrs.pname}-${finalAttrs.version}/* $out/maven
```

Comment:
Does everything need to be copied over?

Reply:
Yes, except perhaps for `LICENSE` / `NOTICE` / `README.txt`, which are present on Arch Linux, anyway.